discovery_engine_core/web-api/src/bin/ingestion.rs

```diff
@@ -283,16 +308,16 @@ fn with_client(client: Client) -> impl Filter<Extract = (Client,), Error = Infal
 }
 
 fn serialize_to_ndjson(
-    documents: &Vec<(String, ElasticDocumentData)>,
+    documents: &Vec<(DocumentId, ElasticDocumentData)>,
 ) -> Result<Bytes, GenericError> {
     let mut bytes = BytesMut::new();
 
     fn write_record(
-        document_id: String,
+        document_id: DocumentId,
         document_data: &ElasticDocumentData,
         bytes: &mut BytesMut,
     ) -> Result<(), GenericError> {
-        let bulk_op_instruction = BulkOpInstruction::new(document_id);
+        let bulk_op_instruction = BulkOpInstruction::new(document_id.0);
```

Comment:
without making inner value public, you could use `as_ref()`, but I'm not sure if cloning here is ok, maybe @rustonaut or @janpetschexain could weigh in on that.

```rust
let id = document_id.as_ref();
let bulk_op_instruction = BulkOpInstruction::new(id.clone());
```

Alternatively we could from `BuilkOpInstruction` in favor of using `json!` makro

```rust
let bulk_op_instruction = serde_json::to_vec(&json!({
  "index": { "_id": document_id.as_ref() }
}))?;
```

Reply:
Instead of making the string in `DocumentId` pub you could implement `From<DocumentId> for String`, this would be stylistically a bit better as it would allow you to turn the `DocumentId` into a `String` but it doesn't allow you to put a malformed string into the `DocumentId` (e.g. `"\0"` or `""`), (And also keep the `as_ref()`.)